File: 2019/01/ruby/fuel.rb
module Fuel
  class << self
    def get_requirement(mass)
      (mass / 3) - 2
    end

    def get_recursive_requirement(mass)
      initial_fuel = get_requirement(mass)
      return 0 if initial_fuel <= 0

      initial_fuel + get_recursive_requirement(initial_fuel)
    end
  end
end

File: 2019/01/python/fuel.py
class Fuel:
  @staticmethod
  def get_requirement(mass: int) -> int:
    return (mass // 3) - 2
    
  @staticmethod
  def get_recursive_requirement(mass: int) -> int:
    initial_fuel = Fuel.get_requirement(mass)

    if initial_fuel <= 0:
      return 0
    
    return initial_fuel + Fuel.get_recursive_requirement(initial_fuel)
